ExecutionResult is None in DetectObjects ROS Action

When using the detect_objects action the ExecutionResult returned by pyrobosim to the pyrobotsim_ros wrapper is None.
```bash
[ERROR] [1741771114.873669517] [pyrobosim.action_server]: Error raised in execute callback: 'NoneType' object has no attribute 'status'
Traceback (most recent call last):
  File "/opt/ros/jazzy/lib/python3.12/site-packages/rclpy/action/server.py", line 333, in _execute_goal
    execute_result = await await_or_execute(execute_callback, goal_handle)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/ros/jazzy/lib/python3.12/site-packages/rclpy/executors.py", line 111, in await_or_execute
    return callback(*args)
           ^^^^^^^^^^^^^^^
  File "/delib_ws/install/pyrobosim_ros/lib/python3.12/site-packages/pyrobosim_ros/ros_interface.py", line 663, in robot_detect_objects_callback
    execution_result=execution_result_to_ros(execution_result),
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/delib_ws/install/pyrobosim_ros/lib/python3.12/site-packages/pyrobosim_ros/ros_conversions.py", line 228, in execution_result_to_ros
    status=getattr(ExecutionStatus, result.status.name),
                                    ^^^^^^^^^^^^^
```
I think the issue stems from the call to [`detect_objects`](https://github.com/sea-bass/pyrobosim/blob/e7f7d54602b8c1b8bdca83b4843113ca46f0d659/pyrobosim/pyrobosim/gui/world_canvas.py#L502) in `world_canvas.py` as this only [starts](https://github.com/sea-bass/pyrobosim/blob/e7f7d54602b8c1b8bdca83b4843113ca46f0d659/pyrobosim/pyrobosim/gui/action_runners.py#L163) the detection but does not return a result.

--- a/pyrobosim_ros/pyrobosim_ros/ros_interface.py
+++ b/pyrobosim_ros/pyrobosim_ros/ros_interface.py
@@ -639,14 +639,9 @@ class WorldROSWrapper(Node):  # type: ignore[misc]
         :param robot: The robot instance corresponding to this request.
         :return: The object detection action result.
         """
-        if self.world.gui is not None:
-            execution_result = self.world.gui.canvas.detect_objects(
-                robot, goal_handle.request.target_object
-            )
-        else:
-            execution_result = robot.detect_objects(
-                target_object=goal_handle.request.target_object
-            )
+        execution_result = robot.detect_objects(
+            target_object=goal_handle.request.target_object
+        )
 
         detected_objects_msg = [
             ObjectState(

